dds/DCPS/XTypes/TypeAssignability.cpp

```diff
@@ -355,41 +379,48 @@ bool TypeAssignability::assignable_struct(const MinimalTypeObject& ta,
 
   // For any member m2 of T2, if there is a member m1 of T1 with the same
   // ID, then the type KeyErased(m1.type) is-assignable-from the type
-  // KeyErased(m2.type)
+  // KeyErased(m2.type).
+  // We extend this rule for all members of T1 and T2 that have the same
+  // ID, i.e., for any pair of members m2 of T2 and m1 of T1 with the same
+  // ID where m1.type and m2.type are not aggregated, verify that
+  // m1.type is-assignable-from m2.type.
```

Comment:
An alternate way of looking at this: "KeyErased(T) = T for any non-aggregated type T"

I think this is what the spec intends, though it may not explicitly say so.

Reply:
I fixed the comment noting this.